### DISCUSSION TITLE
passing a roi mask to MONAILabel.

### DISCUSSION THREAD

**Discussion by dorkylever** [Project-MONAI/MONAILabel#1638] [Q&A]
Hi all, 

This may be a stupid question but is there a way to pass a mask (that specifies an roi) for  training.  I want to see if  I can develop a model that can detect the presence of tumours (liver nets) within the liver and just want to pass it the liver mask so it looks specifically there. 

The simple alternative is to obviously just pass MONAILabel the ROI but I just want to find out if MONAI has masking options. 

Kind Regards, 
Kyle Drover

**Comment by diazandr3s** [comment]
Hi @dorkylever,

Many thanks for posting this question. It is actually a very question.

There is a multi-stage model implementation in MONAI Label that follows the approach you just described. To demonstrate this, we used the spine segmentation task. 

Here are more details of this: https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/radiology#Multistage-Vertebra-Segmentation

In this case, the first stage segments the whole spine, which produces a spine mask—the second and third stage focus on accurately segmenting the vertebrae.

For the liver task, you may not need three stages. Two stages should be enough.

Here is another implementation of this multi-stage approach for teeth segmentation: https://projectweek.na-mic.org/PW37_2022_Virtual/Projects/MultistageTeethSegmentation/

Forked repo:  https://github.com/dgmato/MONAILabel/blob/a903eeb2ac216b9534734703eb0006a6bda09c0b/sample-apps/radiology/lib/infers/dental_pipeline.py

I'd recommend getting familiar with the [pipeline](https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/infers/vertebra_pipeline.py) implementation and the focus on the second stage. The first stage in your case (liver segmentation) is already implemented in the segmentation model: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation.py#L42

I hope this helps,
